homeassistant/components/permobil/config_flow.py

```diff
@@ -0,0 +1,217 @@
+"""Config flow for MyPermobil integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from mypermobil import MyPermobil, MyPermobilAPIException, MyPermobilClientException
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_CODE, CONF_EMAIL, CONF_REGION, CONF_TOKEN, CONF_TTL
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import APPLICATION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+GET_EMAIL_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): cv.string,
```

Comment:
```suggestion
        vol.Required(CONF_EMAIL): vol.Email(),
```

As the schema validates that it is a valid e-mail address, do we still need to call `p_api.set_email(email)` as this function does only validation?

Reply:
`p_api.set_email(email)` also sets the email of the p_api object as a side effect so it is still needed.

Reply:
when i apply the change the config flow stops working, and if i search for `vol.Required(CONF_EMAIL):` in the codebase, every other instance i see the type is set to string and not to vol.email. for example the econet integration.
What should the type be?

Reply:
```suggestion
        vol.Required(CONF_EMAIL): TextSelector(
            TextSelectorConfig(type=TextSelectorType.EMAIL)
        ),
```

Can be used instead. `vol.Email` can be used only in yaml, which is not allowed anymore, my fault

Reply:
Thanks! 
I implemented the textselector based on the Eight Sleep integration!